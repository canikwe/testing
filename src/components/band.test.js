import React from 'react'
//enzyme dependancies 
import { shallow } from 'enzyme'

//component needs 
import Band from './Band'


describe('Band component tests', ()=>{
    let mockProps = {
        id: 1,
        name: "wrm up band",
        deleteBand: jest.fn()
    }
    let wrapper = shallow(<Band {...mockProps} />);
    it('Renders correctly w snapshot', ()=>{
        expect(wrapper).toMatchSnapshot();
    } )

    it("deletes band onClick", () => {
        wrapper.find("button").simulate("click");
        expect(mockProps.deleteBand).toBeCalledWith(mockProps.id)
    })
})