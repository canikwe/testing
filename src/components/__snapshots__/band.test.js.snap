// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Band component tests Renders correctly w snapshot 1`] = `
<div>
  <li>
    wrm up band
  </li>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
